### PR TITLE
Add essential 

### LINKED ISSUE #7: make an essential version!!
~~modcore 2.0~~ essential got released with sk1er beta mods and rpd still uses modcore so make an essential version for all the essentialers out there

### PR DESCRIPTION
**Pull request by Wyvest**
remake build.gradle
add essential (closes #7)
Make Ignore Overlay settings apply instantly
Add the option to retain overlays while randomly selecting a pack

### CODE REVIEW

**Comment by Wyvest** [comment]
oh and ill do something to the rounded stuff

**Comment by Wyvest** [comment]
ok nevermind i dont want to

File: src/main/java/dev/fxe/mods/resourcepackdisplay/data/Config.java [hunk 1]
```diff
@@ -24,81 +24,88 @@
...
+            type = PropertyType.SWITCH, name = "Enable HUD",
+            category = "HUD"
     )
-    public static boolean isGuiEnabled;
+    public static boolean hudEnabled = true;
 
     @Property(
-        type = PropertyType.SWITCH, name = "Display Pack background",
-        category = "HUD"
+            type = PropertyType.SWITCH, name = "Display Pack Background",
```
Comment (1fxe) [inline src/main/java/dev/fxe/mods/resourcepackdisplay/data/Config.java:48]: Continuation Idents should be 4, but not really a big deal

File: settings.gradle [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        maven {
+            name = "Jitpack"
+            url = "https://jitpack.io/"
+        }
+    }
+    resolutionStrategy {
+        eachPlugin {
+            // If the "net.minecraftforge.gradle.forge" plugin is requested we redirect it to asbyth's ForgeGradle fork
+            if (requested.id.id == "net.minecraftforge.gradle.forge") {
+                useModule("com.github.asbyth:ForgeGradle:${requested.version}")
```
Comment (1fxe) [inline settings.gradle:22]: Why? Normal forge works fine

File: build.gradle [hunk 3]
```diff
@@ -54,8 +57,40 @@ processResources {
...
+
+
+    exclude 'LICENSE.md'
+    exclude 'pack.mcmeta'
+    exclude 'dummyThing'
+    exclude '**/module-info.class'
+    exclude '*.so'
+    exclude '*.dylib'
+    exclude '*.dll'
+    exclude '*.jnilib'
```
Comment (1fxe) [inline build.gradle:84]: These don't event get included in the jar anyway? (except META-INF)

**Review by 1fxe — CHANGES_REQUESTED** [review]
Rest seems fine

File: settings.gradle (hunk 2 above)
Comment (Wyvest) [reply settings.gradle:22]: > Why? Normal forge works fine

because normal forge doesn't work with gradle 6

File: build.gradle (hunk 3 above)
Comment (Wyvest) [reply build.gradle:84]: > These don't event get included in the jar anyway? (except META-INF)

oh really? will remove then

File: settings.gradle (hunk 2 above)
Comment (1fxe) [reply settings.gradle:22]: gradle sssssadadasda

**Pull request merged**